Cython/Utility/CppConvert.pyx

```diff
@@ -52,14 +52,24 @@ cdef vector[X] {{cname}}(object o) except *:
 #################### vector.to_py ####################
 
 cdef extern from *:
-    cdef cppclass vector "const std::vector" [T]:
+    cdef cppclass vector "std::vector" [T]:
         size_t size()
         T& operator[](size_t)
 
-@cname("{{cname}}")
-cdef object {{cname}}(vector[X]& v):
-    return [v[i] for i in range(v.size())]
+cdef extern from "Python.h":
+    void Py_INCREF(object)
+    list PyList_New(Py_ssize_t size)
+    void PyList_SET_ITEM(object list, Py_ssize_t i, object o)
 
+@cname("{{cname}}")
+cdef object {{cname}}(const vector[X]& v):
+    o = PyList_New(<Py_ssize_t>v.size())
```

Comment:
It still feels like we should add a max size check here (and also for `cpp_list` further down), because, why not?

Reply:
what is the appropriate exception type?

Reply:
`OverflowError`?

Reply:
Actually, raising a `MemoryError` might be better. After all, we'd be trying to allocate a huge list of Python objects (4-8 byte pointers plus objects), which would obviously fail if already its size exceeds the maximum allowed length. The size check would just prevent the size overflow that might lead to silent data loss instead of an exception.

Basically, the only realistic case where this may occur is a bit vector. Or maybe a `char` vector. Or something like that. Maybe something forged. But we should handle (and reject) such extreme cases correctly. Whenever we cast an unsigned to a signed size, it's worth a maximum size check.